src/framework/actions/turn.ts

```diff
@@ -1,13 +1,120 @@
-import { GameState } from "../types";
+import { CardID, GameState } from "../types";
+import { discardAll, drawCard } from "./hand";
+import { MAX_GOLD } from "../constants";
+
+const handleReadyPhase = ($: GameState): void => {
+  if ($.turnPhase === "READY") {
+    const P = $.players[$.activePlayer];
+
+    P.hasShuffledThisTurn = false;
+    P.hasBuiltWorkerThisTurn = false;
+
+    P.patrol.squadLeader = null;
+    P.patrol.elite = null;
+    P.patrol.scavenger = null;
+    P.patrol.technician = null;
+    P.patrol.lookout = null;
+
+    for (const I of Object.values($.instances)) {
+      if (I.owner === $.activePlayer) {
```

Comment:
nit: consider something like:
```suggestion
    for (const iid of queryInstances($, {
      player: $.activePlayer,
    }) {
      const I = $.instances[iid];
```

Reply:
Good call.